src/shotgunEventDaemon.py

```diff
@@ -507,10 +516,76 @@ def _mainLoop(self):
         while self._continue:
             # Process events
             events = self._getNewEvents()
+            skipPlugins = []
+
             for event in events:
                 for collection in self._pluginCollections:
-                    collection.process(event)
-                self._saveEventIdData()
+                    for plugin in collection:
+                        pluginName = plugin.getName()
+
+                        if pluginName in skipPlugins:
+                            self.log.debug(
+                                "Not processing event #%d. Plugin %s is waiting for a gap to fill.",
+                                event["id"],
+                                pluginName,
+                            )
+                            continue
+
+                        if not plugin.isActive():
+                            self.log.debug(
+                                "Not processing event #%d. Plugin %s is inactive.",
+                                event["id"],
+                                pluginName,
+                            )
+                            continue
+
+                        nextEventId = plugin.getNextUnprocessedEventId()
+
+                        if nextEventId is None:
+                            msg = (
+                                "Plugin %s has no last event marker. Sending event #%d."
```

Comment:
Should it be more clear to the user what an event marker is?

Reply:
Changing to `Plugin %s has not yet processed any events. Sending event #%d.`